efficientdet/main.py

```diff
@@ -254,56 +243,36 @@ def _can_partition(spatial_dim):
   model_fn_instance = det_model_fn.get_model_fn(FLAGS.model_name)
   max_instances_per_image = config.max_instances_per_image
   eval_steps = int(FLAGS.eval_samples // FLAGS.eval_batch_size)
+  total_examples = int(config.num_epochs * FLAGS.num_examples_per_epoch)
+  train_steps = total_examples // FLAGS.train_batch_size
   use_tpu = (FLAGS.strategy == 'tpu')
   logging.info(params)
 
-  def _train(steps):
-    """Build train estimator and run training if steps > 0."""
-    train_estimator = tf.estimator.tpu.TPUEstimator(
-        model_fn=model_fn_instance,
-        use_tpu=use_tpu,
-        train_batch_size=FLAGS.train_batch_size,
-        config=run_config,
-        params=params)
-    train_estimator.train(
-        input_fn=dataloader.InputReader(
-            FLAGS.training_file_pattern,
-            is_training=True,
-            use_fake_data=FLAGS.use_fake_data,
-            max_instances_per_image=max_instances_per_image),
-        max_steps=steps)
-
-  def _eval(steps):
-    """Build estimator and eval the latest checkpoint if steps > 0."""
-    eval_params = dict(
-        params,
-        strategy=FLAGS.strategy,
-        input_rand_hflip=False,
-        is_training_bn=False,
-    )
-    eval_estimator = tf.estimator.tpu.TPUEstimator(
-        model_fn=model_fn_instance,
-        use_tpu=use_tpu,
-        train_batch_size=FLAGS.train_batch_size,
-        eval_batch_size=FLAGS.eval_batch_size,
-        config=run_config,
-        params=eval_params)
-    eval_results = eval_estimator.evaluate(
-        input_fn=dataloader.InputReader(
-            FLAGS.validation_file_pattern,
-            is_training=False,
-            max_instances_per_image=max_instances_per_image),
-        steps=steps,
-        name=FLAGS.eval_name)
-    logging.info('Evaluation results: %s', eval_results)
-    return eval_results
+  # Use the unified estimator, train, and eval interfaces.
+  estimator = tf.estimator.tpu.TPUEstimator(
+      model_fn=model_fn_instance,
+      use_tpu=use_tpu,
+      train_batch_size=FLAGS.train_batch_size,
+      eval_batch_size=FLAGS.eval_batch_size,
+      config=run_config,
+      params=params)
+  train_input_fn = dataloader.InputReader(
+      FLAGS.training_file_pattern,
+      is_training=True,
+      use_fake_data=FLAGS.use_fake_data,
+      max_instances_per_image=max_instances_per_image)
+  eval_input_fn = dataloader.InputReader(
+      FLAGS.validation_file_pattern,
+      is_training=False,
+      use_fake_data=FLAGS.use_fake_data,
+      max_instances_per_image=max_instances_per_image)
 
   # start train/eval flow.
   if FLAGS.mode == 'train':
-    total_examples = int(config.num_epochs * FLAGS.num_examples_per_epoch)
-    _train(total_examples // FLAGS.train_batch_size)
+    total_examples = int(config.num_epochs * FLAGS.num_examples_per_epoch),
```

Comment:
Could I remove the line?

Reply:
Oh, yes, this line should be deleted!
I saw you already fixed in the other PR. Thanks for the fix!